web/src/javascript/app.js

```diff
@@ -3,15 +3,26 @@ import {drawGame, createGameCanvas} from "./game";
 import {gameRefresh} from "./constants";
 
 const gameCanvas = createGameCanvas();
-let currentState;
 
-(new Worker("javascript/gameWebWorker.bundle.js")).onmessage = message => currentState = message.data;
+let currentState;
+const networkWorker = new Worker("javascript/gameWebWorker.bundle.js");
+networkWorker.onmessage = message => {
+  // This is to prevent Chrome's GC from deleting the worker.
+  // It's not happening on FF, I'm pretty sure it's a V8 bug.
+  // Also, it could be caused by a mix of V8 and webpack/babel.
+  if(!networkWorker) {
+    console.error("Got GCed");
+  }
+  currentState = message.data;
+};
 
 async function updateLoop() {
+  let startTime = (new Date()).valueOf();
   if(currentState) {
     drawGame(currentState, gameCanvas);
     drawLeaderboard(currentState);
   }
+  setTimeout(updateLoop, 1000/gameRefresh - ((new Date()).valueOf()) - startTime);
```

Comment:
I don't think the calculation is doing what is expected here.

The goal is to do something like:
```
wait for (how_long_I_want_to_wait_per_frame - how_long_I've_been_processing)
```
Right? We're doing:
```
sleep(refresh_time - current_time - previous_time)
```
Which is not the same as:
```
sleep(refresh_time - elapsed_time)
```
Because:
```
elapsed_time = current_time - start_time
refresh_time - elapsed_time = refresh_time - (current_time - start_time)
                            = refresh_time - current_time + start_time
                            != refresh_time - current_time - start_time
```

I think it would be clearer if we added a variable in there (and that would fix it):
```js
const elapsed = new Date().getTime() - startTime
setTimeout(updateLoop, 1000/gameRefresh - elapsed);
```
Thoughts?